### DISCUSSION TITLE
Mysterious pops and clicks in FLAC: SOLVED!

### DISCUSSION THREAD

**Discussion by Vseason** [rern/rAudio#1225] [Fixed Issues]
For anyone who may have encountered pops and clicks in their FLAC files over the past few months, even in albums that previously played without issue, I can report that a lack of padding in FLAC tags for the affected files *appears* to be the explanation. When FLAC audio was tagged without any padding, and with artwork solely present as an external file, I experienced numerous, non-repeatable (meaning not always in the same place) pops and clicks during playback, but simply re-tagging the affected tracks with padding enabled, and artwork embedded rather than solely external, resolved all audible playback issues.

The audio data in the padded and unpadded FLAC files is bit-identical, according to foobar2000. I do not have any clue about what changed in the underlying libraries that would explain the problematic behavior for unpadded FLAC tags, but am documenting it here, should anyone else encounter similar issues when attempting to save space.

**Comment by R101G** [comment]
Which Pi are you using?

**Reply by Vseason** [reply]
I’m using a Raspberry Pi 3B+.

**Reply by R101G** [reply]
Interesting. I wonder if it is the same click problem as I am getting when the local browser is disabled.

**Reply by Vseason** [reply]
On my own system the local browser is always enabled, so your description of the issue does not seem to match my experience, unless you were *also* noticing such clicks exclusively when playing FLAC files tagged without padding or embedded artwork.

Out of curiosity, how are your audio_buffer and max_output_buffer configured?
